audio_service/example/lib/main.dart

```diff
@@ -1,63 +1,189 @@
-// ignore_for_file: public_member_api_docs
-
-// FOR MORE EXAMPLES, VISIT THE GITHUB REPOSITORY AT:
-//
-//  https://github.com/ryanheise/audio_service
-//
-// This example implements a minimal audio handler that renders the current
-// media item and playback state to the system notification and responds to 4
-// media actions:
-//
-// - play
-// - pause
-// - seek
-// - stop
-//
-// To run this example, use:
-//
-// flutter run
-
 import 'dart:async';
+import 'dart:math';
 
 import 'package:audio_service/audio_service.dart';
-import 'package:audio_service_example/common.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:just_audio/just_audio.dart';
+import 'package:audio_session/audio_session.dart';
 import 'package:rxdart/rxdart.dart';
 
-// You might want to provide this using dependency injection rather than a
-// global variable.
-late AudioHandler _audioHandler;
+late AudioPlayerHandler _audioHandler;
 
 Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
   _audioHandler = await AudioService.init(
     builder: () => AudioPlayerHandler(),
-    config: const AudioServiceConfig(
-      androidNotificationChannelId: 'com.ryanheise.myapp.channel.audio',
-      androidNotificationChannelName: 'Audio playback',
+    config: AudioServiceConfig(
+      androidNotificationChannelName: 'Audio Service Demo',
       androidNotificationOngoing: true,
+      preloadArtwork: true,
+      // preloadArtwork: false,
     ),
   );
-  runApp(const MyApp());
+  runApp(App());
 }
 
-class MyApp extends StatelessWidget {
-  const MyApp({Key? key}) : super(key: key);
+class AudioPlayerHandler extends BaseAudioHandler
+    with QueueHandler, SeekHandler {
+  AudioPlayerHandler() {
+    _init();
+  }
+
+  final _player = AudioPlayer();
+  final items = [
+    for (int i = 0; i < 3; i++)
+      MediaItem(
+        id: '$i',
+        album: '$i',
+        title: 'fallback null $i',
+      ),
+    for (int i = 3; i < 6; i++)
+      MediaItem(
+        id: '$i',
+        album: '$i',
+        title: 'fallback bad network uri $i',
+        artUri: Uri.parse("https://www.google.com/"),
+      ),
+    for (int i = 6; i < 9; i++)
+      MediaItem(
+        id: '$i',
+        album: '$i',
+        title: 'fallback bad content uri $i',
+        artUri: Uri.parse("content://123"),
+      ),
+    for (int i = 9; i < 12; i++)
+      MediaItem(
+        id: '$i',
+        album: '$i',
+        title: 'fallback bad file uri $i',
+        artUri: Uri.parse("file://123"),
+      ),
+    for (int i = 12; i < 500; i++)
+      MediaItem(
+        id: '$i',
+        album: '$i',
+        title: 'other uris $i',
+        artUri: Uri.parse("https://picsum.photos/900/900?random=$i"),
+      ),
+  ];
```

Comment:
This is a temporary demonstration of the changes in this PR. I can come up with something that we can actually put into example a bit later, if you want.

These items contain 4 cases of fallback URIs:
1. artUri = null
2. invalid network URI
3. invalid file URI
4. invalid content URI

The remaining items are valid